### ISSUE TITLE
Is ebm3/thermalexpansion misnamed?

### ISSUE THREAD

**Issue by bobkopp** [radical-collaboration/facts#338]
@victor-malagon @AlexReedy is this module misnamed? It looks like perhaps it also does the regional dynamic sea level calculation, which means it should be named sterodynamics and not thermalexpansion. 

Victor, we also need the calibration of this module to be replicable and documented before this goes into main.

**Comment by AlexReedy** [comment]
oh maybe so yea!

Also @bobkopp if you're talking about the calibrated constraint parameter files they can be found here:
https://zenodo.org/records/11506798

**Comment by victor-malagon** [comment]
Hi @bobkopp,

Yes, I agree. It would be more correct to call this module sterodynamics. It was first called thermalexpansion when we worked on just the global projections, and then for the regional we built on this existing module, but it should have been renamed. 

As it is structured right now, the project script does global projections, and postprocess does regional projections. This is not ideal as it does not follow previous conventions. Should we split this process following the structure of other modules? Including: preprocess, fit, project, and postprocess? And saving intermediate data in pickle files?

I also downloaded and preprocessed a few more cmip6 zos simulations. These new simulations are especially important for long-term (2300) projections. Once, this module is finished and ready to go to main, we should also update the cmip6 data folder.

**Comment by bobkopp** [comment]
I'm not wedded to the four fold division, but if the postprocess step actually has no dependencies on the project step, it might make sense to actually publish these as two modules (ebm3/thermalexpansion and ebm3/oceandynamics) - tlm is one module because the ocean dynamic position depends upon the output of the thermal expansion calculation, but I am not sure that's the case here. 

In any case, it seems like you and @AlexReedy should have an architectural discussion. 

The most important point from my perspective is that everything be replicable.

**Comment by victor-malagon** [comment]
That's not the case here. To compute sterodynamics, we do take thermal expansion from the previous step to add it to Dynamic Sea Level, but apart from that both steps have no dependencies. Making two separate modules seems like a good option, i.e., we run only ebm3/thermalexpansion for global, and both ebm3/thermalexpansion and ebm3/oceandynamics for regional. 

I'll discuss this with @AlexReedy

**Comment by bobkopp** [comment]
@victor-malagon Can we come to a resolution here so this can be cleaned up for the 1.2 release?

**Comment by victor-malagon** [comment]
@bobkopp I am planning to clean up and finish all PROTECT-related modules in mid-September. Would that be in time for the 1.2 release?

**Comment by bobkopp** [comment]
@victor-malagon Yes, that's great. Our goal is for all the PROTECT updates to be in 1.2.

**Comment by victor-malagon** [comment]
@bobkopp I have been making some updates on the sterodynamics sea level module. Since we discussed with Tamsin the possibility to include overshooting scenarios, I have also been working on implementing that with available CMIP6 simulations. Thus, it has taking me a bit longer than anticipated. I'll be busy over the next two weeks but expect to have a finalised module in October. I'll keep you and @AlexReedy updated on this.

**Comment by kemccusker** [comment]
closed by #378
